### ISSUE TITLE
`ape compile` doesn't detect updates to dependencies of source files

### ISSUE THREAD

**Issue by fubuloubu** [ApeWorX/ape#623]
### Environment information

* `ape` and plugin versions:

```
$ ape --version
0.1.4.dev3+ge9615911

$ ape plugins list
Installed Plugins:
  solidity    
  hardhat     
  vyper    
```

* Python Version: 3.9.9
* OS: linux

### What went wrong?

Updating a file did not trigger re-compiling vs. the cached copy

### How can it be fixed?

Fill this in if you have ideas on how the bug could be fixed.

**Comment by pandadefi** [comment]
here is an example from https://github.com/yearn/veYFI
First compile the code:
```
git checkout 657459797cdec60ee1739b9f63cb7b484569c4b3
ape compile -f
```

Second:
```
git checkout 1dd38049149a782b87308968593834b2ae2fbd2f
ape test
```

The test are falling because the contracts aren't re-compiled and missing https://github.com/yearn/veYFI/commit/1dd38049149a782b87308968593834b2ae2fbd2f#diff-730c6bf6b59b69b8d287c6a118eb5e741a229272167aca3e8a931ff2a0e14049R86

**Comment by fubuloubu** [comment]
We are pretty sure we narrowed this down to an issue resolving updated dependencies of files that produced relevant types, which is how `Gauge.sol` is using `BaseGauge.sol` right now in your project

**Comment by NotPeopling2day** [comment]
Depends on #704

**Comment by NotPeopling2day** [comment]
Solution for solidity contracts is merged, this ticket is superseded by implementation tickets in vyper and cairo repos.
